### ISSUE TITLE
NNBD. 'e is Null' promotion doesn't work

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#41494]
According to the NNBD [specification](https://github.com/dart-lang/language/blob/master/accepted/future-releases/nnbd/feature-specification.md#null-promotion)

> A check of the form e == null or of the form e is Null where e has static type T promotes the type of e to Null in the true continuation, and to NonNull(T) in the false continuation.

But check `e is Null` doesn't work nor in Analyzer nor in CFE

```dart
class A {
  foo() {}
}

dynamic init(x) => x;

main() {
  A? a = init(A());

  if (a == null) {
  } else {
    a.foo();     // Ok
    A a1 = a;    // Ok
  }

  if (a is Null) {
  } else {
    a.foo();    // Error: Method 'foo' cannot be called on 'A?' because it is potentially null.
    A a1 = a;   // Error: A value of type 'A?' can't be assigned to a variable of type 'A'
  }
}
```

Tested on `Dart VM version: 2.8.0-dev.20.0 (dev) (Fri Apr 3 10:19:55 2020 +0200) on "windows_x64"`

**Comment by vsmenon** [comment]
fyi - @eernstg - looks like bug in both, correct?

**Comment by eernstg** [comment]
@vsmenon wrote:
> looks like bug in both

Agreed, and it looks like @johnniwinther agrees as well.

**Comment by scheglov** [comment]
This was fixed in eada4d8d652e5ca05f165e1e293064056b5ebbba.
